corral/util/log.pony

```diff
@@ -99,16 +19,35 @@ primitive CodeLogFormatter is LogFormatter
      .> append(": ")
      .> append(msg)
 
-primitive LevelLogFormatter is LogFormatter
-  fun apply(level: LogLevel, msg: String, loc: SourceLoc): String =>
-    (recover String(level.string().size()
-      + msg.size()
-      + 4)
-    end)
-     .> append(level.string())
-     .> append(": ")
-     .> append(msg)
+primitive LevelLogFormatter
+  """
+  LevelLogFormatter produces LogFormatters that prepend a log level to entries.
```

Comment:
This doesn't really feel like explains the purpose.

We want to have the log level in all the output for filtering, searching etc that folks might want to do with an unstructured log.

Rather than adding the format to each string at all call locations, we create one of these in main that in turn wraps the other formatters appending the log-level"

That's more than the comment for this primitive, but part of that should be included in this primitive.

I'd like to see log.pony moved from corral/util to corral/log or corral/logging and have a package level doc as well that explains the relationship between this logger and the others and how logging is in general used in corral (as detailed above).

Including the discussion about the short circuit that your commit switched to.

Basically cover everything that I felt the need to ask you about as part oft this PR.

Reply:
Thank you for the clarification. I've added some more clarification. Let me know what you think.